coderedcms/models/page_models.py

```diff
@@ -533,6 +608,24 @@ def get_context(self, request, *args, **kwargs):
         )
         return context
 
+    def get_mini_render_template(self):
+        """
+        Get the template for use with the mini 'preview' render
+        """
+        return self.mini_view_template
+
+    def render_mini_view(self, request, *args, **kwargs):
+        """
+        Render this page a small 'preview' component, suitable for inclusion on other
+        pages.
+
+        Set mini_view_template on this model to override what template is used, or
+        override coderedcms/templates/pages/mini_view_page.html to change all instances.
+        """
+        ctx = self.get_context(request=request, *args, **kwargs)
+        template_name = self.get_mini_render_template()
+        return render_to_string(template_name, ctx, request)
```

Comment:
Not sure if I like this, it feels a bit backwards returning a string of rendered HTML..

I would instead opt to override `get_context` and include the queryset of related pages in the context. Then add a block in the base.html template for related content (similar to how we do for showing search results https://github.com/coderedcorp/coderedcms/blob/dev/coderedcms/templates/coderedcms/pages/search.html#L67).

Reply:
This is for rendering itself as a mini view. This is the generic ability to render itself as a mini preview, for part two of the ticket.